Added suuport for a docker-registry VM/LB/Autoscaling Group

jimconner opened the pull request:
Two additional files added to the terraform config to provision a LaunchConfiguration, AutoScaling Group and ELB for a resilient docker-registry instance.

The AutoScaling group ensures that we will always have exactly 1 docker-registry server running.
The LoadBalancer is required for a consistent IP address (and later for additional health checks)

Commit:
Added suuport for a docker-registry VM/LB/Autoscaling Group

mikkoc commented on docker-registry.tf:
```diff
@@ -0,0 +1,44 @@
+  */
+}
+
+/* Docker-registry internal Load balancer */
```
It seems we're missing the `internal = true` option. See: https://www.terraform.io/docs/providers/aws/r/elb.html#internal

mikkoc commented on docker-registry.tf:
```diff
@@ -0,0 +1,44 @@
+/* Docker-registry internal Load balancer */
+resource "aws_elb" "docker-registry" {
+  name = "docker-registry"
+  subnets = ["${aws_subnet.public1.id}", "${aws_subnet.public2.id}"]
```
These probably need to be private subnets, unless the registry needs to be accessible from the outside world.

mikkoc commented on docker-registry.tf:
```diff
@@ -0,0 +1,44 @@
+resource "aws_elb" "docker-registry" {
+  name = "docker-registry"
+  subnets = ["${aws_subnet.public1.id}", "${aws_subnet.public2.id}"]
+  security_groups = ["${aws_security_group.default.id}", "${aws_security_group.web.id}"]
```
The web security group is meant to be for publicly accessible services. We need to create an additional security group that allows access from within the cluster, only on port 5000.

mikkoc commented:
It'd be nice to add a Route53 DNS record to point to the ELB name, so we don't need to update the Ansible configuration whenever we update the ELB and the name changes.

Commit:
Added changes which were suggested by Mikko

docker-registry port changed to 6000 - thanks Nassim

Did not add new security group suggested by Mikko as 'default'
security group already allows all traffic within the VPC.

dcarley commented:
> The AutoScaling group ensures that we will always have exactly 1 docker-registry server running.

Based on discussions today about auto-scaled machines not being bootstrapped by Ansible, would it be preferable to not use an autoscaling group for this, at least for the time being and while it's a single node?

dcarley commented:
From discussion IRL: we decided that the autoscaling group is complexity that we don't need at the moment because we can only have one instance of the registry and we don't have any integration between autoscaling groups and Ansible (the machine will be recreated but not configured). Jim (and maybe Luigi) are going refactor it down to a single instance and a CNAME (not autoscaling group or ELB).

Commit:
Trimmed down docker-registry deployment down to just a single VM

- Removed Autoscaler Group
- Removed Load Balancer
- Removed Launch Configuration
+ Added single VM instance

Commit:
An untested attempt at adding DNS record for docker-registry

Commit:
Changed to a CNAME for docker-registry

Commit:
Fixed identifier for docker-registry dns entry

Commit:
Changed back to an A record as CNAME wasn't working
